autobuild/mac/artifacts/autobuild_mac_2_build.sh

```diff
@@ -16,4 +21,28 @@ source "$(dirname "${BASH_SOURCE[0]}")/../../ensure_THIS_JAMULUS_PROJECT_PATH.sh
 cd "${THIS_JAMULUS_PROJECT_PATH}"
 
 echo "Run deploy script..."
-sh "${THIS_JAMULUS_PROJECT_PATH}"/mac/deploy_mac.sh
+
+# If we have certificate details, then prepare the signing
+if [[ -z "${MACOS_CERTIFICATE_PWD}" ||
+     -z "${MACOS_CERTIFICATE}" ||
+     -z "${MACOS_CERTIFICATE_ID}" ||
+     -z "${NOTARIZATION_PASSWORD}" ||
+     "${SIGN}" != "sign_if_possible" ]]; then
+    sh "${THIS_JAMULUS_PROJECT_PATH}"/mac/deploy_mac.sh
+else
+    echo "Setting up signing, as all credentials found"
+
+    # Get the cert to a file
+    echo $MACOS_CERTIFICATE | base64 --decode > certificate.p12
+
+    # Set up a keychain for the build
+    security create-keychain -p T3mpP4ssword build.keychain
```

Comment:
`T3mpP4ssword` --> although it's probably only valid for this run, can't you use either an unique ID or a GH secret here?

Reply:
that would make it more secure. I'll add.